### PR TITLE
Saleor 4019 add with choice field to attribute

### PR DESCRIPTION
**Pull request by dexon44**
I want to merge this change because it add witch choice field.

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [x] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-4019 Add withChoices field to Attribute](https://app.clickup.com/t/2549495/SALEOR-4019)

File: saleor/graphql/attribute/tests/queries/test_attribute_query.py [hunk 1]
```diff
@@ -573,3 +574,51 @@ def test_attributes_in_collection_query(
...
+    ],
+)
+def test_attributes_withChoice_flag(
+    user_api_client,
+    input_type,
+    expected_withChoice_return,
+):
+    attribute = Attribute.objects.create(
+        slug="boolean",
+        name="Boolean",
```
Comment (maarcingebala) [inline saleor/graphql/attribute/tests/queries/test_attribute_query.py:598]: We could use more accurate values here, maybe generate them from input type?

File: saleor/graphql/attribute/tests/queries/test_attribute_query.py [hunk 2]
```diff
@@ -573,3 +574,51 @@ def test_attributes_in_collection_query(
...
+        (AttributeInputType.DROPDOWN, True),
+        (AttributeInputType.MULTISELECT, True),
+        (AttributeInputType.FILE, False),
+        (AttributeInputType.REFERENCE, False),
+        (AttributeInputType.NUMERIC, False),
+        (AttributeInputType.RICH_TEXT, False),
+        (AttributeInputType.BOOLEAN, False),
+    ],
+)
+def test_attributes_withChoice_flag(
```
Comment (maarcingebala) [inline saleor/graphql/attribute/tests/queries/test_attribute_query.py:591]:
```suggestion
def test_attributes_with_choice_flag(
```

**Review by maarcingebala — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -128,6 +128,7 @@ All notable, unreleased changes to this project will be documented in this file.
 - Modify order of auth middleware calls - #7572 by @tomaszszymanski129
 - Add app support for events - #7622 by @IKarbowiak
 - Add date & date time attributes - #7500 by @piotrgrundas
+- Add withChoice flag for Attribute Type - #7733 by @dexon44
```
Comment (maarcingebala) [inline CHANGELOG.md:131]: 🚓 
```suggestion
- Add `withChoices` flag for Attribute type - #7733 by @dexon44
```

File: saleor/graphql/attribute/tests/queries/test_attribute_query.py [hunk 4]
```diff
@@ -573,3 +574,51 @@ def test_attributes_in_collection_query(
     expected_flat_attributes_data = list(expected_qs.values_list("slug", flat=True))
 
     assert flat_attributes_data == expected_flat_attributes_data
+
+
+@pytest.mark.parametrize(
+    "input_type, expected_withChoice_return",
```
Comment (IKarbowiak) [inline saleor/graphql/attribute/tests/queries/test_attribute_query.py:580]:
```suggestion
    "input_type, expected_with_choices_value",
```

File: saleor/graphql/attribute/tests/queries/test_attribute_query.py [hunk 5]
```diff
@@ -573,3 +574,51 @@ def test_attributes_in_collection_query(
...
+        (AttributeInputType.REFERENCE, False),
+        (AttributeInputType.NUMERIC, False),
+        (AttributeInputType.RICH_TEXT, False),
+        (AttributeInputType.BOOLEAN, False),
+    ],
+)
+def test_attributes_with_choice_flag(
+    user_api_client,
+    input_type,
+    expected_withChoice_return,
```
Comment (IKarbowiak) [inline saleor/graphql/attribute/tests/queries/test_attribute_query.py:594]: Same here 👮‍♀️

**Review by IKarbowiak — APPROVED** [review]

**Pull request merged**